### PR TITLE
Feature/update based on comments round 2

### PR DESCRIPTION
**Pull request by Nauthiz-Jera**

### CODE REVIEW

File: spec/alchemy_spec.rb [hunk 1]
```diff
@@ -8,23 +8,47 @@
...
         end 
         httpMockInstance = HttpMockInstance.new(false)
-    }
+    end
 
     it "can call fetchTransacions and return hash, from, and to" do
         http = double("http")
         
+        httpMockInstance.request_response = Struct.new(:read_body).new(
+            "{\"jsonrpc\":\"2.0\",\"id\":null,\"result\":{\"blockHash\":\"0x1d59ff54b1eb26b013ce3cb5fc9dab3705b415a67127a003c3e61eb445bb8df2\",\"blockNumber\":\"0x5daf3b\",\"hash\":\"0x88df016429689c079f3b2f6ad39fa052532c56795b733da78a91ebe6a713944b\",\"from\":\"0xa7d9ddbe1f17865597fbd27ec712455208b6b76d\",\"gas\":\"0xc350\",\"gasPrice\":\"0x4a817c800\",\"input\":\"0x68656c6c6f21\",\"nonce\":\"0x15\",\"r\":\"0x1b5e176d927f8e9ab405058b2d2457392da3e20f328b16ddabcebc33eaac5fea\",\"s\":\"0x4ba69724e8f69de52f0125ad8b3c5c2cef33019bac3249e2c0a2192766d1721c\",\"to\":\"0xf02c1c8e6114b1dbe8937a39260b5b0a374432bb\",\"transactionIndex\":\"0x41\",\"type\":\"0x0\",\"v\":\"0x25\",\"value\":\"0xf3dbb76162000\"}}"
```
Comment (Nauthiz-Jera) [inline spec/alchemy_spec.rb]: Still need to fix this and make it cleaner. The reason it is a json string here is that it's how we get it back from the block chain, but will work on a cleaner way of doing this.

File: bin/setup [hunk 2]
```diff
@@ -4,5 +4,6 @@ IFS=$'\n\t'
 set -vx
 
 bundle install
+bundle lock --add-platform x86_64-linux
```
Comment (Nauthiz-Jera) [inline bin/setup:7]: Adding this line fix the CI

Comment (seansu4you87) [reply bin/setup:7]: oh nice, what did this do / what was the issue?

Comment (Nauthiz-Jera) [reply bin/setup:7]: In the gemfile.lock it set linux 64 bit OS as one of the platforms. It seems like that when you install ruby through rbenv or possibly just ruby in general it defaults to the 64 bit version of Ruby.

Comment (Nauthiz-Jera) [reply bin/setup:7]: Looking back at the failed pipeline we can see it trying to use 64 bit version as well:
```
Error: Error: The process '/opt/hostedtoolcache/Ruby/3.1.2/x64/bin/bundle' failed with exit code 16
    at ExecState._setResult (/home/runner/work/_actions/ruby/setup-ruby/v1/dist/index.js:5381:25)
    at ExecState.CheckComplete (/home/runner/work/_actions/ruby/setup-ruby/v1/dist/index.js:5[36](https://github.com/Nauthiz-Jera/ruby-technical/actions/runs/3889361925/jobs/6637556899#step:3:43)4:18)
    at ChildProcess.<anonymous> (/home/runner/work/_actions/ruby/setup-ruby/v1/dist/index.js:5258:27)
    at ChildProcess.emit (node:events:390:28)
    at maybeClose (node:internal/child_process:1064:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:301:5)
```

Comment (seansu4you87) [reply bin/setup:7]: Oh nice, looks like you got it working now =]

**Pull request merged**